app/routes/github.webhook/pull-request-webhook.ts

```diff
@@ -13,25 +13,65 @@ export async function processPullRequestWebhook(
   repository: Repository
 ) {
   if (!repository.deployOnlyOnPullRequest)
-    return json({ message: "No action taken" });
+    return json({
+      status: "skipped",
+      message: "No action taken",
+      reason: "Repository is not configured for PR-only deployments",
+      repository: repository.id
+    });
 
   const [files, branch] = await findPullRequestData();
 
-  if (!branch && !hasChangesAffectingNextjsApp())
-    return json({ message: "No action taken" });
+  if (!branch) {
+    return json({
+      status: "skipped",
+      message: "No action taken",
+      reason: "No branch found",
+      repository: repository.id
+    });
+  }
+
+  if (!hasChangesAffectingNextjsApp())
+    return json({
+      status: "skipped",
+      message: "No action taken",
+      reason: "No changes affecting the Next.js application",
+      pullRequest: webhook.payload.number,
+      repository: repository.id
+    });
```

Comment:
This changed, but I _think_ this is actually what we want, right?

If there's no branch, we can't queue up a build. If there's no changed files, we don't want to queue up a build. They are independent.

Reply:
"branch" here comes from a DB find for previous builds.

Reply:
Nvm, this isn't what we want.